frontend/src/components/unsubscribe/Unsubscribe.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react'
+import { useParams } from 'react-router-dom'
+import querystring from 'querystring'
+
+import { ErrorBlock, PrimaryButton, TextButton } from 'components/common'
+
+import styles from './Unsubscribe.module.scss'
+import appLogo from 'assets/img/brand/app-logo.svg'
+import landingHero from 'assets/img/unsubscribe/request-unsubscribe.png'
+import cancelRequestHero from 'assets/img/unsubscribe/cancel-request.png'
+
+import { unsubscribeRequest } from 'services/unsubscribe.service'
+
+const Unsubscribe = () => {
+  const { version } = useParams()
+  const [errorMsg, setErrorMsg] = useState('')
+  const [isUnsubscribed, setUnsubscribed] = useState(false)
+  const [isStaying, setStaying] = useState(false)
+
+  async function onConfirmation() {
+    try {
+      const query = new URL(window.location.href).searchParams.toString()
+      const params = querystring.parse(query)
+
+      // Check to make sure that the search params are all strings and not []strings
+      const isValid = Object.values(params).every((v) => typeof v === 'string')
+      if (!isValid) {
+        throw new Error(
+          'Search params should all be strings and not array of strings.'
+        )
+      }
+
+      const { c: campaignId, r: recipient, h: hash } = params
+
+      await unsubscribeRequest({
+        campaignId: +campaignId,
+        recipient: recipient as string,
+        hash: hash as string,
+        version,
+      })
+      setUnsubscribed(true)
+    } catch (err) {
+      setErrorMsg(err.message)
+    }
```

Comment:
When a request with invalid query params is rejected by the Joi validator for the `/unsubscribe` request, the error message returned is `"c" must be a number`. We can simplify the error displayed here like I mentioned above.